src/py/pyodide/console.py

```diff
@@ -405,18 +405,21 @@ def done_cb(fut: asyncio.Task[Any]) -> None:
                 res.set_result(fut.result())
             res = None
 
-        ensure_future(self.runcode(source, code)).add_done_callback(done_cb)
+        ensure_future(self._runcode_inner(source, code)).add_done_callback(done_cb)
         return res
 
+    async def _runcode_inner(self, source: str, code: CodeRunner) -> Any:
```

Comment:
`_runcode_inner` calling `runcode` seems strange to me namingwise. How about `_runcode_with_lock` or `_runcode_atomic` or something like that, explicitly mentioning that it acquires lock?

Reply:
Done. Renamed it to `_runcode_with_lock`. Choosing an appropriate name is so difficult. That's one of the reason why I like using decorators so much — this way I won't need to think of another function name 😂